openfasoc/generators/gdsfactory-gen/process_input.py

```diff
@@ -0,0 +1,233 @@
+				syntaxes.append("create/define [a/an] var_type variable called/named varname =/equal valorexpr")#variable
+				syntaxes.append("save/dump [conversation/code] to pathtosaveto\n\tNOTE: save code is assumed if neither conversation nor code are specified")#
+				# print all
+				self.print_to_stream("\nBelow are valid sentences:")
+				#for i, syntax in enumerate(syntaxes):
+				for syntax in syntaxes:
+					#self.print_to_stream(str(i)+" : "+syntax)
+					self.print_to_stream(syntax)
+				self.print_to_stream()
+			elif mode_indicator[0]=="i":# import
+				imports = sentence.replace("and",",").replace("import","").replace("from","").strip().split(",")
+				for modimport in imports:
+					if modimport=="" or modimport.isspace():
+						continue
+					words = modimport.strip().split()
+					compname = words[0]
+					modpath = words[1] if len(words)>1 else None
+					aliases = [compname]#TODO implement comp aliasing
+					self.code.update_import_table(aliases,compname,modpath)
+			elif "create" in mode_indicator or "define" in mode_indicator:# create/define a variable or param
+				vartype = None
+				for word in words:
+					word = word.lower().strip()
+					if "int" in word:
+						vartype = int
+					elif "float" in word:
+						vartype = float
+					elif "bool" in word:
+						vartype = bool
+					elif "str" in word:
+						vartype = str
+					elif "tupl" in word:
+						vartype = tuple
+					if vartype is not None:
+						break
+				varname = None
+				for i, word in enumerate(words):
+					if word in ["called","named"]:
+						varname = words[i+1]
+						break
+				expr = None
+				eqpar = sentence.replace("equal","=").strip().removesuffix(".")
+				expr = eqpar.split("=")[1] if "=" in eqpar else None
+				if "parameter" in words:
+					self.code.update_parameter_table(varname,vartype,None,None)
+				else:#variable
+					self.code.update_variable_table(varname,expr)
+			elif mode_indicator[0]=="p":# place
+				if "configuration" in sentence:
+					raise NotImplementedError("configs not yet implemented")
+				genid = self.code.find_first_generator_id(sentence)
+				comp_name = None
+				for i, word in enumerate(words):
+					if word in ["called","named"]:
+						comp_name = words[i+1]
+						break
+				params = None
+				for i, word in enumerate(words):
+					if word=="with":
+						params = " ".join(words[i:])
+						break
+				self.code.update_place_table(genid,params,comp_name)
+			elif mode_indicator[0]=="r":# route
+				port1 = None
+				port2 = None
+				routetype = None
+				params = None
+				for i, word in enumerate(words):
+					if word in ["between","from"]:
+						port1 = words[i+1] if port1 is None else port1
+					elif word in ["from","and"]:
+						port2 = words[i+1] if port2 is None else port2
+					elif word in ["using","a","an"]:
+						routetype=words[i+1]
+					elif word=="with":
+						params = " ".join(words[i:])
+				self.code.update_route_table(port1,port2,params,routetype)
+			elif mode_indicator[0]=="m":# move
```

Comment:
Again, having a separate method for each of the commands and calling them based on the command given would be better for readability and debugging.